Description: Fix FTBFS
Author: Roger Light <roger@atchoo.org>
Forwarded: yes
--- a/src/bridge.c
+++ b/src/bridge.c
@@ -228,7 +228,7 @@
 {
 	int rc;
 
-	rc = net__socket_connect_step3(context, context->bridge->addresses[context->bridge->cur_address].address, context->bridge->addresses[context->bridge->cur_address].port, NULL, false);
+	rc = net__socket_connect_step3(context, context->bridge->addresses[context->bridge->cur_address].address);
 	if(rc > 0){
 		if(rc == MOSQ_ERR_TLS){
 			net__socket_close(db, context);
